c_emulator/riscv_sim.cpp

```diff
@@ -991,17 +782,18 @@ int main(int argc, char **argv)
     exit(1);
   }
 
-  do {
+  if (!rvfi) {
     run_sail();
-#ifndef RVFI_DII
-  } while (0);
-#else
-    if (rvfi_dii) {
-      /* Reset for next test */
-      reinit_sail(entry);
-    }
-  } while (rvfi_dii);
-#endif
+  } else {
+    do {
+      run_sail();
+      if (rvfi) {
+        /* Reset for next test */
+        reinit_sail(entry);
+      }
+    } while (rvfi);
+  }
```

Comment:
This should be the same as just the second part:

```
    do {
      run_sail();
      if (rvfi) {
        /* Reset for next test */
        reinit_sail(entry);
      }
    } while (rvfi);
```
no need for the if else anymore.